File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if(k==0)
        {
            return;
        }
        int n = nums.size();
        vector <int> temp;
        k = k % n;
        
        for(int i = n-k; i < n; i++)
        {
            temp.push_back(nums[i]);
        }
        for(int i = 0; i < n-k; i++)
        {
            temp.push_back(nums[i]);
        }
        for(int i = 0; i < n; i++)
        {
            nums[i] = temp[i];
        }
    }
};

File: 189-rotate-array/rotate-array.java
class Solution {
    void Reverse (int[] arr,int start,int end)
    {
        while (start <= end)
        {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k%n;
        Reverse(nums,0,n-k-1);
        Reverse(nums,n-k,n-1);
        Reverse(nums,0,n-1);

    }
}